src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,594 @@
+                    }
+
+                    // Process headers
+                    // Cors has it's own restrictions on headers.
+                    // https://developer.mozilla.org/en-US/docs/Web/API/Headers
+                    using (HostObject jsHeaders = new HostObject("Headers"))
+                    {
+                        if (request.Headers != null)
+                        {
+                            foreach (KeyValuePair<string, IEnumerable<string>> header in request.Headers)
+                            {
+                                foreach (string value in header.Value)
+                                {
+                                    jsHeaders.Invoke("append", header.Key, value);
+                                }
+                            }
+                        }
+                        if (request.Content?.Headers != null)
+                        {
+                            foreach (KeyValuePair<string, IEnumerable<string>> header in request.Content.Headers)
+                            {
+                                foreach (string value in header.Value)
+                                {
+                                    jsHeaders.Invoke("append", header.Key, value);
+                                }
+                            }
+                        }
+                        requestObject.SetObjectProperty("headers", jsHeaders);
+                    }
+
+                    WasmHttpReadStream? wasmHttpReadStream = null;
+
+                    JSObject abortController = new HostObject("AbortController");
+                    JSObject signal = (JSObject)abortController.GetObjectProperty("signal");
+                    requestObject.SetObjectProperty("signal", signal);
+                    signal.Dispose();
+
+                    CancellationTokenSource abortCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                    CancellationTokenRegistration abortRegistration = abortCts.Token.Register((Action)(() =>
+                    {
+                        if (abortController.JSHandle != -1)
+                        {
+                            abortController.Invoke("abort");
+                            abortController?.Dispose();
+                        }
+                        wasmHttpReadStream?.Dispose();
+                    }));
+
+                    var args = new Interop.JavaScript.Array();
+                    if (request.RequestUri != null)
+                    {
+                        args.Push(request.RequestUri.ToString());
+                        args.Push(requestObject);
+                    }
+
+                    requestObject.Dispose();
+
+                    var response = s_fetch?.Invoke("apply", s_window, args) as Task<object>;
+                    args.Dispose();
+                    if (response == null)
+                        throw new Exception("Internal error marshalling the response Promise from `fetch`.");
```

Comment:
We need to ensure abortCts is disposed of.

Reply:
Addressed in recent commit

Reply:
This is addressed in recent commits.